src/TwigComponent/src/Command/TwigComponentDebugCommand.php

```diff
@@ -21,27 +21,29 @@
 use Symfony\Component\Console\Output\OutputInterface;
 use Symfony\Component\Console\Style\SymfonyStyle;
 use Symfony\Component\Finder\Finder;
-use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
 use Symfony\UX\TwigComponent\ComponentFactory;
 use Symfony\UX\TwigComponent\ComponentMetadata;
-use Symfony\UX\TwigComponent\Twig\PropsNode;
+use Symfony\UX\TwigComponent\ComponentPropertiesExtractor;
 use Twig\Environment;
 use Twig\Loader\FilesystemLoader;
 
 #[AsCommand(name: 'debug:twig-component', description: 'Display components and them usages for an application')]
 class TwigComponentDebugCommand extends Command
 {
     private readonly string $anonymousDirectory;
+    private readonly ComponentPropertiesExtractor $componentPropertiesExtractor;
 
     public function __construct(
         private string $twigTemplatesPath,
         private ComponentFactory $componentFactory,
         private Environment $twig,
         private readonly array $componentClassMap,
         ?string $anonymousDirectory = null,
+        ?ComponentPropertiesExtractor $componentPropertiesExtractor = null,
     ) {
         parent::__construct();
         $this->anonymousDirectory = $anonymousDirectory ?? 'components';
+        $this->componentPropertiesExtractor = $componentPropertiesExtractor ?? new ComponentPropertiesExtractor($this->twig);
```

Comment:
If we can create it like this, why injecting it ? 

If we need to inject it, then we should trigger a deprecation if not provided here